ledger-service/http-json/src/itlib/scala/http/AbstractWebsocketServiceIntegrationTest.scala

```diff
@@ -262,51 +262,167 @@ abstract class AbstractWebsocketServiceIntegrationTest
   "interface sub" - {
     "query endpoint" in withHttpService { fixture =>
       import fixture.uri
-      val query = """[
+      val query =
+        """[
         {"templateIds": ["IAccount:IAccount"], "query": {"isAbcPrefix": true}},
         {"templateIds": ["IAccount:IAccount"], "query": {"is123Suffix": true}}
       ]"""
 
-      def createAccount(
+      @nowarn("msg=pattern var evtsWrapper .* is never used")
+      def resp(
           owner: domain.Party,
-          amount: String,
           headers: List[HttpHeader],
-      ) = postCreateCommand(
-        accountCreateCommand(owner, amount),
-        fixture,
-        headers = headers,
-      )
+          kill: UniqueKillSwitch,
+      ): Sink[JsValue, Future[ShouldHaveEnded]] = {
+        def createAccount(
+            owner: domain.Party,
+            amount: String,
+            headers: List[HttpHeader],
+        ) = postCreateCommand(
+          accountCreateCommand(owner, amount),
+          fixture,
+          headers = headers,
+        )
+
+        def exerciseTransferPayload(cid: domain.ContractId) = {
+          import json.JsonProtocol._
+          Map(
+            "templateId" -> "IAccount:IAccount".toJson,
+            "contractId" -> cid.toJson,
+            "choice" -> "ChangeAmount".toJson,
+            "argument" -> Map("newAmount" -> "abcxx").toJson,
+          ).toJson
+        }
+
+        val dslSyntax = Consume.syntax[JsValue]
+        import dslSyntax._
+
+//        def readAndExtract(amount: String): Consume.FCC[JsValue, CreatedAccountEvent] = for {
```

Comment:
I have tried to reduce duplicated code here by what @S11001001 suggested. I have got it complied but IntelliJ doesn't like it with a lot of red error here. I wonder if I can make IntelliJ happier

Reply:
Making IntelliJ happier is like separating out the red M&Ms. Less red, but ultimately pointless because they all taste the same anyhow.